src/app/static/src/js/components/runWorkflow/runWorkflowReport.vue

```diff
@@ -444,8 +449,16 @@ export default Vue.extend<Data, Methods, Computed, Props>({
         stepIsValid(newVal) {
             this.$emit("valid", newVal);
         },
-        reportsOrigin(newVal) {
+        reportsOrigin: async function (newVal) {
             session.setSelectedWorkflowReportSource(newVal);
+
+            const newlyActiveReports = this.inactiveOriginWorkflowReports;
+            this.inactiveOriginWorkflowReports = [...this.workflowMetadata.reports];
+            await this.updateWorkflowReports(newlyActiveReports);
+            session.setInactiveOriginWorkflowReports(this.inactiveOriginWorkflowReports);
+
+            // Revalidate reports as branch may have changed while other origin was active
+            this.validateWorkflowReportsWithGit()
```

Comment:
```suggestion
            await this.activateReports(newVal)
```

Reply:
It would be more readable to separate the code.

```
async activateReports (newVal) {
            session.setSelectedWorkflowReportSource(newVal);

            const activatedReports = this.deactivatedReports;
            this.deactivatedReports = [...this.workflowMetadata.reports];
            await this.updateWorkflowReports(activatedReports);
            session.setInactiveOriginWorkflowReports(this.deactivatedReports);

            // Revalidate reports as branch may have changed while other origin was active
            await this.validateWorkflowReportsWithGit()
        }
```

Reply:
Really? That seems less readable to me, since it's just moving the entire lengthy code chunk to a new method, so the maintaining developer has to go and hunt that down and deal with the whole thing there rather than just seeing the code changes in the watch. I'd be more convinced about, say, pulling out the section which swaps inactive and active into a separate method and giving that a meaningful name - I'll add that, and you can see what you think.